src/PostgREST/App.hs

```diff
@@ -291,9 +291,13 @@ handleRead headersOnly identifier context@RequestContext{..} = do
             )
           ]
           ++ contentTypeHeaders context
+        rsOrErrBody = if status == HTTP.status416
+          then Error.errorPayload $ Error.ApiRequestError $ ApiRequestTypes.InvalidRange
+            $ ApiRequestTypes.OutOfBounds (show $ RangeQuery.rangeOffset iTopLevelRange) (maybe "0" show total)
```

Comment:
Giving this a second look. It's a really long function chain.. could this logic be abstracted in Errors.hs?